### PR TITLE
8411 Edit draft financing statement.

### PR DESCRIPTION
**Pull request by doug-lovett**
Signed-off-by: Doug Lovett <doug@diamante.ca>

*Issue #:* /bcgov/entity#8411

*Description of changes:*
Implement my registrations table edit draft button click event. Load draft details in the store and navigate to the first step in a new financing statement registration.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/utils/registration-helper.ts [hunk 1]
```diff
@@ -183,6 +185,87 @@ export async function saveDischarge (stateModel:StateModelIF): Promise<Discharge
...
+    console.error('getDraft failed: response null.')
+    return stateModel
+  }
+  if (draft !== undefined && draft.error !== undefined) {
+    console.error('getDraft failed: ' + draft.error.statusCode + ': ' + draft.error.message)
+    return stateModel
+  }
+
+  var registrationType:RegistrationTypeIF = null
+  for (const regType of RegistrationTypes) {
```
Comment (jordiwes) [inline ppr-ui/src/utils/registration-helper.ts]: wondering if you can do a find here?
 RegistrationTypes.find(obj => {
    return obj.registrationTypeAPI ===  draft.financingStatement.type
  })

File: ppr-ui/tests/unit/RegistrationHelper.spec.ts [hunk 2]
```diff
@@ -131,6 +137,28 @@ describe('Registration API Helper Tests', () => {
...
+    expect(stateModel.registration.parties.registeringParty).toBeDefined()
+    expect(stateModel.registration.lengthTrust).toBeDefined()
+    expect(stateModel.registration.lengthTrust.lifeYears).toBe(5)
+    expect(stateModel.registration.registrationType.registrationTypeAPI).toBe(APIRegistrationTypes.SECURITY_AGREEMENT)
+    expect(stateModel.registration.parties.securedParties).toBeDefined()
+    expect(stateModel.registration.parties.securedParties.length).toBe(1)
+    expect(stateModel.registration.parties.debtors).toBeDefined()
+    expect(stateModel.registration.parties.debtors.length).toBe(1)
+    expect(stateModel.registration.collateral.vehicleCollateral).toBeDefined()
+    expect(stateModel.registration.collateral.vehicleCollateral.length).toBe(1)
```
Comment (jordiwes) [inline ppr-ui/tests/unit/RegistrationHelper.spec.ts:161]: perhaps we could check the current store values as well?

**Review by thorwolpert — APPROVED** [review]
nothing much to add to Steph's comments

File: ppr-ui/src/utils/registration-helper.ts (hunk 1 above)
Comment (doug-lovett) [reply ppr-ui/src/utils/registration-helper.ts]: Great idea, I made the change.

File: ppr-ui/tests/unit/RegistrationHelper.spec.ts (hunk 2 above)
Comment (doug-lovett) [reply ppr-ui/tests/unit/RegistrationHelper.spec.ts:161]: Also a good idea, I have added some store checks.

**Review by jordiwes — APPROVED** [review]
Great!

**Comment by jordiwes** [comment]
test fail though?

**Pull request merged**